### PR TITLE
fix(app): Add gripper error handling and fix bugs

### PR DESCRIPTION
**Pull request by smb2268**
# Overview

This PR adds error handling to gripper calibration and fixes a few wizard styling and copy bugs, screenshots and video below!

# Test Plan
Run through gripper calibration flow
Ensure pipette single/8-channel attach copy is correct
Look at spacing on 96 channel flow on ODD

# Changelog
1. Change mount back to `LEFT` from `EXTENSION` for gripper flows since extension is behaving weirdly
2. Add error handling for calibration errors in gripper flows
3. Update copy for single and 8-channel mount pipette step
4. Fix spacing issue for ODD wizard screens where title overflows to second line

# Review requests
Review screenshots & code

# Risk assessment
Low

Screen Shot 2023-07-18 at 4 20 01 PM
Screen Shot 2023-07-18 at 5 20 13 PM

https://github.com/Opentrons/opentrons/assets/14302493/f5e1663f-a784-42b7-9c0f-7fedd60e8231

### CODE REVIEW

File: app/src/organisms/GripperWizardFlows/MovePin.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import * as React from 'react'
-import { useTranslation } from 'react-i18next'
-import { EXTENSION } from '@opentrons/shared-data'
+import { useTranslation, Trans } from 'react-i18next'
+import { LEFT } from '@opentrons/shared-data'
+import { COLORS, TYPOGRAPHY } from '@opentrons/components'
 import { css } from 'styled-components'
```
Comment (koji) [inline app/src/organisms/GripperWizardFlows/MovePin.tsx]:
```suggestion
import { css } from 'styled-components'
import { LEFT } from '@opentrons/shared-data'
import { COLORS, TYPOGRAPHY } from '@opentrons/components'
```

File: app/src/organisms/GripperWizardFlows/index.tsx [hunk 2]
```diff
@@ -66,6 +66,10 @@ export function GripperWizardFlows(
     },
   })
   const [isExiting, setIsExiting] = React.useState<boolean>(false)
+  const [errorMessage, setShowErrorMessage] = React.useState<string | null>(
```
Comment (koji) [inline app/src/organisms/GripperWizardFlows/index.tsx]: If there is no specific reason to use `setShowErrorMessage`
```suggestion
  const [errorMessage, setErrorMessage] = React.useState<string | null>(
```

**Review by koji — COMMENTED** [review]
Seems that a couple of components are missing `errorMessage` and `setShowErrorMessage`.

**Comment by koji** [comment]
Screenshot 2023-07-18 at 6 32 57 PM

missing currentStep/totalStep text
also spinner screen needs it too.

alignment of Title and Step 4/4 is a little bit off.
Screenshot 2023-07-18 at 6 41 42 PM

File: app/src/organisms/GripperWizardFlows/MovePin.tsx [hunk 3]
```diff
@@ -201,11 +214,34 @@ export const MovePin = (props: MovePinProps): JSX.Element | null => {
...
+        }
+        alternativeSpinner={errorMessage == null ? inProgressImage : undefined}
       />
     )
-  return (
+  return errorMessage != null ? (
+    <SimpleWizardBody
+      isSuccess={false}
+      iconColor={COLORS.errorEnabled}
+      header={t('shared:error_encountered')}
```
Comment (koji) [inline app/src/organisms/GripperWizardFlows/MovePin.tsx:227]: In this branch, shared.json doesn't have `error_encountered`. pipette_wizard_flows.json has it.
I think moving `error_encountered` to shared.json would be good.

Screenshot 2023-07-18 at 6 49 57 PM

**Comment by koji** [comment]
@smb2268 
One question about images you posted.

For border-radius, the first image and the second image looks different. I guess the second one is using 4px.
Also the header's vertical alignment (tittle + step X/Y) is different.

The styling is not applied to GenericWizardTile, but applied separately?

File: app/src/organisms/GripperWizardFlows/MovePin.tsx (hunk 3 above)
Comment (smb2268) [reply app/src/organisms/GripperWizardFlows/MovePin.tsx:227]: Now that I changed the base branch to edge this should be resolved. `error_encountered` is in `shared.json` in edge.

**Comment by smb2268** [comment]
@koji the issues you posted about above are due to most of the Gripper DQA being completed on edge while this PR was initially against internal release. I rebased this to be off of edge so there should be fewer conflicts and those UI bugs should go away.

File: app/src/molecules/GenericWizardTile/index.tsx [hunk 4]
```diff
@@ -51,10 +52,10 @@ const GO_BACK_BUTTON_DISABLED_STYLE = css`
 `
 const Title = styled.h1`
   ${TYPOGRAPHY.h1Default};
-
   @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
     ${TYPOGRAPHY.level4HeaderSemiBold};
     height: ${SPACING.spacing40};
+    display: inline-block;
```
Comment (jerader) [inline app/src/molecules/GenericWizardTile/index.tsx]: i'm assuming you can use the `DISPLAY_INLINE_BLOCK` here

**Review by jerader — APPROVED** [review]
logic looks sound to me, left a comment but don't wanna block this from merging!

File: app/src/organisms/InstrumentInfo/index.tsx [hunk 5]
```diff
@@ -118,51 +118,57 @@ export const InstrumentInfo = (props: InstrumentInfoProps): JSX.Element => {
...
+            <MediumButton
+              buttonType="secondary"
+              flex="1"
+              onClick={handleDetach}
+              buttonText={t('detach')}
+              textTransform={TYPOGRAPHY.textTransformCapitalize}
+              justifyContent={JUSTIFY_CENTER}
+            />
+            <MediumButton
+              buttonType="primary"
```
Comment (koji) [inline app/src/organisms/InstrumentInfo/index.tsx]:
```suggestion

```

File: app/src/organisms/InstrumentInfo/index.tsx [hunk 6]
```diff
@@ -118,51 +118,57 @@ export const InstrumentInfo = (props: InstrumentInfoProps): JSX.Element => {
...
+              flex="1"
+              onClick={handleRecalibrate}
+              buttonText={
+                instrument.data.calibratedOffset?.last_modified == null
+                  ? t('calibrate')
+                  : t('recalibrate')
```
Comment (koji) [inline app/src/organisms/InstrumentInfo/index.tsx:163]: recommend to use i18n.format with `capitalize`

**Review by koji — APPROVED** [review]
lgtm
left a couple of comments but they are small things and we can refactor later.

**Pull request merged**